### DISCUSSION TITLE
How to intercept avatar image request?

### DISCUSSION THREAD

**Discussion by zeroarst** [GetStream/stream-chat-android#5267] [Q&A] [answered]
In iOS SDK, it provides `ImageCDN` allowing developers to intercept the request. We use this to add tokens to the header if the url is our host:
```
import Foundation
import StreamChatSwiftUI


class ChatImageCDN: StreamImageCDN {
    override func urlRequest(forImage url: URL) -> URLRequest {
        var request = URLRequest(url: url)

        // FIXME: Dont use hardcoded host.
        if url.host == "my.company.com" {
            // Add auth token to request headers.
        }

        return request
    }
}
```

But in android, I don't see anywhere I can do this.
Bascially all avatar images are pointing to our own server, but requies access token.
Can you suggest how to do that? Is it possible?

**Comment by JcMinarro** [comment, marked as answer]
Hello @zeroarst 
The SDK allows you to include an instance of [ImageHeadersProvider](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/ChatUI.kt#L70) that is used when pictures are fetched.

Recently we updated this interface to allow you to decide which URLs need header injection and which do not. https://github.com/GetStream/stream-chat-android/pull/5265
It will be available in the next release
